opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -1913,10 +1913,11 @@ export const elPaginate = async (context, user, indexName, options = {}) => {
     );
 };
 export const elList = async (context, user, indexName, options = {}) => {
-  const { first = MAX_SEARCH_SIZE, infinite = false } = options;
+  const { first = MAX_SEARCH_SIZE, maxSize = undefined, infinite = false } = options;
```

Comment:
is `infinite` option used somewhere ? I can't find it anywhere in the code except here in this method.

There is also a very similar method `elList` in `middleware-loader.ts` , with exactly the same logic, can we deduplicate ?

Reply:
Nice catch, will take a look

Reply:
So i refactor to keep only 1 implementation. To do that and prevent circle dependencies i was forced to create engine-loader (like middleware-loader) to start extract some logic and convert that to TS. Due to time limitation i dont really spend some time to correctly convert to TS.

Reply:
I can have a look at that !

Reply:
Wow. 1k+ lines of code. It might take a while 😅

Reply:
Its more code move than implementation :)

Reply:
This is definitely a good start, I don't see much _eslint-disable_ and the like... the next stage would be to actually convert the _any_ everywhere.

I started to dig deeper in that direction, to check the impact (see my comments in the code).
IMO as long as converting the any requires changes in the code logic, it's too risky to do it _on the side_, not worth it.
The engine code is too critical to do this lightly.

What we could do iteratively though, is to handle the types locally, where we use `elPaginate` and the like.
We would cast the `any` returned by our engine, and locally the code would be proper TS with a real added value.
This seems feasible...

What do you think ?

Reply:
Yes i think its the right direction. The target of this PR is to deduplicate code and not resolving the TS migration of engine.js.